python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -314,50 +328,143 @@ cdef class Column:
-        responsible for ensuring the data is not mutated unexpectedly while the
-        column is in use.
+            A 1D or 2D list column, depending on the input shape.
+
+        Raises
+        ------
+        ValueError
+            If the object is not 1D or 2D, or is not C-contiguous.
+        ImportError
+            If CuPy is not installed.
         """
-        data = gpumemoryview(obj)
-        iface = data.__cuda_array_interface__
-        if iface.get('mask') is not None:
-            raise ValueError("mask not yet supported.")
+        if np is None or cp is None:
+            raise ImportError("NumPy and CuPy must be installed to use this method")
 
-        typestr = iface['typestr'][1:]
+        iface = getattr(obj, "__cuda_array_interface__", None)
+        if iface is None:
+            raise TypeError("Object does not implement __cuda_array_interface__")
+
+        if iface.get("mask") is not None:
+            raise ValueError("mask not yet supported")
+
+        typestr = iface["typestr"][1:]
         data_type = _datatype_from_dtype_desc(typestr)
 
-        if not is_c_contiguous(
-            iface['shape'],
-            iface['strides'],
-            size_of(data_type)
-        ):
+        shape = iface["shape"]
+        if not is_c_contiguous(shape, iface["strides"], size_of(data_type)):
             raise ValueError("Data must be C-contiguous")
 
-        size = iface['shape'][0]
-        return Column(
-            data_type,
-            size,
-            data,
-            None,
-            0,
-            0,
-            []
+        if len(shape) == 1:
+            data = gpumemoryview(obj)
+            size = shape[0]
+            return cls(data_type, size, data, None, 0, 0, [])
+        elif len(shape) == 2:
+            flat_data = cp.ravel(obj)
+            num_rows, num_cols = shape
+            if num_rows < INT_MAX:
+                offsets_col = sequence(
+                    num_rows + 1,
+                    Scalar.from_numpy(np.int32(0)),
+                    Scalar.from_numpy(np.int32(num_cols))
+                )
+            else:
+                raise ValueError(
+                    "Number of rows exceeds int32 limit for offsets column."
+                )
+            data_col = cls(
+                data_type=data_type,
+                size=flat_data.size,
+                data=gpumemoryview(flat_data),
+                mask=None,
+                null_count=0,
+                offset=0,
+                children=[],
+            )
+            return cls(
+                data_type=DataType(type_id.LIST),
+                size=num_rows,
+                data=None,
+                mask=None,
+                null_count=0,
+                offset=0,
+                children=[offsets_col, data_col],
+            )
+        else:
+            raise ValueError("Only 1D or 2D arrays are supported")
+
+    @classmethod
+    def from_arraylike(cls, obj):
```

Comment:
Curious, are you planning on supporting an "arraylike" like a Python `list` with this API as well?

If you're planning on constraining this to interface objects, I may suggest just `from_array`